src/Promitor.Agents.Scraper/AzureMonitorClientFactory.cs

```diff
@@ -43,17 +44,29 @@ public AzureMonitorClient CreateIfNotExists(AzureEnvironment cloud, string tenan
         private static AzureMonitorClient CreateNewAzureMonitorClient(AzureEnvironment cloud, string tenantId, string subscriptionId, MetricSinkWriter metricSinkWriter, IRuntimeMetricsCollector metricsCollector, IConfiguration configuration, IOptions<AzureMonitorLoggingConfiguration> azureMonitorLoggingConfiguration, ILoggerFactory loggerFactory)
         {
             var azureCredentials = DetermineAzureCredentials(configuration);
-            var azureMonitorClient = new AzureMonitorClient(cloud, tenantId, subscriptionId, azureCredentials.ApplicationId, azureCredentials.Secret, azureMonitorLoggingConfiguration, metricSinkWriter, metricsCollector, loggerFactory);
+            var azureMonitorClient = new AzureMonitorClient(cloud, tenantId, subscriptionId, azureCredentials.AuthenticationMode, azureCredentials.ManagedIdentityId, azureCredentials.ApplicationId, azureCredentials.Secret, azureMonitorLoggingConfiguration, metricSinkWriter, metricsCollector, loggerFactory);
             return azureMonitorClient;
         }
 
         private static AzureCredentials DetermineAzureCredentials(IConfiguration configuration)
         {
+            var authenticationConfiguration = configuration.GetSection("authentication").Get<AuthenticationConfiguration>();
+
+            // To be still compatible with existing infrastructure using previous version of Promitor, we need to check if the authentication section exists.
+            // If not, we should use a default value
+            if (authenticationConfiguration == null)
+            {
+                authenticationConfiguration = new AuthenticationConfiguration();
```

Comment:
We might want to add a logging statement here that they should explicitly configure it through a warning?

Reply:
Oh, indeed, a log warning can be a good idea. Or maybe just a log information since it will still working even if the user does not specify the authentication mode he wants to use.
Eventually this warning (or info) will concern only the user that will upgrade Promitor to the last version, with an already in place configuration

Reply:
Yes, but I'd use a warning since it's higher than info so people will certainly see it - Make sense?